modules/tinymce/src/plugins/image/main/ts/api/Commands.ts

```diff
@@ -9,6 +9,7 @@ import { Dialog } from '../ui/Dialog';
 import Editor from 'tinymce/core/api/Editor';
 
 const register = (editor: Editor) => {
+  // eslint-disable-next-line @typescript-eslint/no-misused-promises
```

Comment:
This looks to be incorrect. The `open` function shouldn't be returning a promise, it should be `void` I believe?

Reply:
`Dialog(editor).open` is of type `() => Promise<void>` so what's wrong?

Reply:
I assume the Promise resolves when the dialog finally opens, however commands just kick things off, they don't care when they complete. So ignoring the promise seems the right thing to do to me...

Reply:
The `open` API for this is normally always type `() => void` so, yes this implementation is technically inconsistent/wrong.

Reply:
It's not public API, it's an implementation detail (if Andy hasn't erased the history you'd see that I implemented it)... So what is wrong here? It was also one of the first dialogs implemented so surely that means that everything else is "wrong"... :-P

Reply:
I haven't erased anything, github just doesn't use `--follow` when showing file history. It's one of the biggest things we lost when we switched.

Reply:
The fact that a) eslint is complaining and b) the return type is `Promise<void>` meaning it's useless shows that this shouldn't be returning a promise as it's not useful. For reference I checked every other plugin and their implementation is all `open: () => void` or `open: (editor: Editor) => void`, so your statement about everything else being wrong is misstated.

@TheSpyder I think James meant the squash commit for 5.0 dev preview, so we need the old history from stash to find out who actually made changes.

Reply:
If there's a reason for it to be async it's not useless. Let's not fight over trivial details, what's the best solution here?

Reply:
The return value is never used, so yeah it's not providing any benefit. The reason it's likely returning anything is probably due to our tendency to do something like `open: () => openWindow()` when we don't actually care about the return value, so it should technically be `open: () => { openWindow(); }` which is a fix made elsewhere in this PR. The exact issue is the following line:
```
const open = () => collect(editor).then(makeDialog(helpers)).then((spec) => {
    editor.windowManager.open(spec);
  });
```
as `collect` rightfully so should be async, but do to using the above mentioned syntax then it actually ends up returning the resulting promise.

Reply:
that does seem like an accident of syntax, so regardless of who did what when I think adding the braces to make it an actual void return is the right approach.

Reply:
The reason it was written like that was that it was possible to open the dialog and wait for it to open. As you've noted collect must be async.

That open returns void is purely convention. In this case the convention did not make sense.

Reply:
We don't ever wait for the promise though? You'd rather disable this warning than remove a capability we never use?

Reply:
I added a descriptive wrapper function to allow ignoring the promise and fixed the original function so the promise is more useful and outputs the DialogApi.
I hope this makes everyone happy.